File: src/com/fusenetworks/fuse/Fuse.java
package com.fusenetworks.fuse;

import com.fusenetworks.fuse.commands.CMD_Handler;
import com.fusenetworks.fuse.commands.CMD_Loader;
import com.fusenetworks.fuse.exploits.Potion;
import com.fusenetworks.fuse.exploits.Sign;
import com.fusenetworks.fuse.listener.*;
import com.fusenetworks.fuse.util.NLog;
import com.fusenetworks.fuse.util.NUtil;
import org.bukkit.Bukkit;
import org.bukkit.Server;
import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.bukkit.plugin.java.JavaPlugin;
import org.bukkit.scheduler.BukkitRunnable;
import org.mcstats.Metrics;

import java.io.File;
import java.io.IOException;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class Fuse extends JavaPlugin {

    public static Fuse plugin;
    public static Server server;
    public static Fuse instance;


    public static String buildDate = "2/10/18";
    public static String buildCreator = "Telesphoreo";
    File jarFile = this.getFile();
    
    @Override
    public void onLoad() {
        Fuse.plugin = this;
        Fuse.server = plugin.getServer();
        NLog.setServerLogger(server.getLogger());
        NLog.setServerLogger(server.getLogger());
    }

    @Override
    public void onEnable() {
        server.getPluginManager().registerEvents(new AutoUpdate(), Fuse.plugin);
        server.getPluginManager().registerEvents(new CommandBlocker(), Fuse.plugin);
        server.getPluginManager().registerEvents(new Commandspy(), Fuse.plugin);
        server.getPluginManager().registerEvents(new ConfigListener(), Fuse.plugin);
        server.getPluginManager().registerEvents(new Developer(), Fuse.plugin);
        server.getPluginManager().registerEvents(new Launchpads(), Fuse.plugin);
        server.getPluginManager().registerEvents(new LoginMessages(), Fuse.plugin);
        server.getPluginManager().registerEvents(new NoFall(), Fuse.plugin);
        server.getPluginManager().registerEvents(new NoHunger(), Fuse.plugin);
        server.getPluginManager().registerEvents(new Potion(), Fuse.plugin);
        server.getPluginManager().registerEvents(new PotionListener(), Fuse.plugin);
        server.getPluginManager().registerEvents(new Sign(), Fuse.plugin);
        Config.loadConfigs();
        new BukkitRunnable() {
            @Override
            public void run() {
                CMD_Loader.getCommandMap();
                CMD_Loader.scan();
            }
        }.runTaskLater(plugin, 20L);
        Bukkit.getScheduler().scheduleSyncRepeatingTask(Bukkit.getPluginManager().getPlugin("Fuse"), () -> {
            if ((NUtil.NEntityWiper.wipeEntities(true, true)) == 1)
            {
                NLog.info((NUtil.NEntityWiper.wipeEntities(true, true)) + " entity removed");
            } else if ((NUtil.NEntityWiper.wipeEntities(true, true)) != 0) {
                NLog.info((NUtil.NEntityWiper.wipeEntities(true, true)) + " entities removed");
            }
        }, 1L , (long) 300 * 20);
    instance = this;
    
    try {
    Metrics metrics = new Metrics(this);
    metrics.start();
    } catch (IOException e) {
        // Failed to submit the stats :-(
    }
    
    }

    @Override
    public void onDisable() {
    }

    public static Fuse getInstance() {
        return instance;
    }

    @Override
    public boolean onCommand(CommandSender sender, Command cmd, String commandLabel, String[] args) {
        return CMD_Handler.handleCommand(sender, cmd, commandLabel, args);
    }
}

File: src/com/fusenetworks/fuse/Fuse.kt
package com.fusenetworks.fuse

import com.fusenetworks.fuse.commands.*
import com.fusenetworks.fuse.commands.Commandspy
import com.fusenetworks.fuse.listener.*
import com.fusenetworks.fuse.listener.Developer
import com.fusenetworks.fuse.util.NLog
import com.fusenetworks.fuse.util.NUtil
import org.bukkit.Bukkit
import org.bukkit.Server
import org.bukkit.plugin.java.JavaPlugin

class Fuse : JavaPlugin() {
    internal var jarFile = this.file

    override fun onLoad() {
        plugin = this
        Fuse.server = plugin!!.server
        NLog.setServerLogger(server.logger)
        NLog.setServerLogger(server.logger)
    }

    override fun onEnable() {
        Bukkit.getPluginManager().registerEvents(AutoUpdate, this);
        Bukkit.getPluginManager().registerEvents(CommandBlocker(), this);
        Bukkit.getPluginManager().registerEvents(Commandspy, this);
        Bukkit.getPluginManager().registerEvents(ConfigListener(), this);
        Bukkit.getPluginManager().registerEvents(Developer, this);
        Bukkit.getPluginManager().registerEvents(Launchpads(), this);
        Bukkit.getPluginManager().registerEvents(LoginMessages(), this);
        Bukkit.getPluginManager().registerEvents(NoFall, this);
        Bukkit.getPluginManager().registerEvents(NoHunger, this);
        Bukkit.getPluginManager().registerEvents(PotionListener(), this);
        Bukkit.getPluginManager().registerEvents(SignPatch, this);
        getCommand("adminchat").executor = Adminchat
        getCommand("admininfo").executor = Admininfo
        getCommand("clearlag").executor = Clearlag
        getCommand("commandspy").executor = Commandspy
        getCommand("consolesay").executor = Consolesay
        getCommand("contributors").executor = Contributors
        getCommand("developer").executor = com.fusenetworks.fuse.commands.Developer
        getCommand("discord").executor = Discord
        Config.loadConfigs()
        Bukkit.getScheduler().scheduleSyncRepeatingTask(Bukkit.getPluginManager().getPlugin("Fuse"), {
            if (NUtil.NEntityWiper.wipeEntities(true, true) == 1) {
                NLog.info(NUtil.NEntityWiper.wipeEntities(true, true).toString() + " entity removed")
            } else if (NUtil.NEntityWiper.wipeEntities(true, true) != 0) {
                NLog.info(NUtil.NEntityWiper.wipeEntities(true, true).toString() + " entities removed")
            }
        }, 1L, 300.toLong() * 20)
        instance = this
    }

    override fun onDisable() {}

    companion object {
        var plugin: Fuse? = null
        var server: Server? = null
        var instance: Fuse? = null


        var buildDate = "9/2/17"
        var buildCreator = "Telesphoreo"
    }
}